### PR TITLE
Make table generation conflicts easier to debug (with colors!)

### PR DESCRIPTION
**Pull request by hughesjj**
[Related issue](https://github.com/open-telemetry/build-tools/issues/120)

# Context
Produces a visual difference in the output to make an error message involved in table generation more contextually meaningful and easier to debug.

I use ANSI color escape codes that [should be compatible](https://en.wikipedia.org/wiki/ANSI_escape_code#Platform_support) between any "modern" terminal in windows, mac, or linux.  The below screenshots were taken on linux using `wezterm`.  I've done my best to make the mechanism for coloring diffs maintainable and understandable. []()

Currently, I output the entire difflib output.  For enhancements on large markdown files, we could consider replacing the "equals" opcode case with some "relative offset" information.  However that's complicated and likely warrants further discussion, so for now I propose we display the full output.  The full output is supposed to be human readable anyway, so it shouldn't be *too* long in practice.

## Current error message
image

## Error message after this review
image

### CODE REVIEW

File: semantic-conventions/src/opentelemetry/semconv/templating/markdown/__init__.py [hunk 1]
```diff
@@ -37,6 +39,73 @@
 from opentelemetry.semconv.templating.markdown.options import MarkdownOptions
 
 
+class VisualDiffer:
```
Comment (joaopgrassi) [inline semantic-conventions/src/opentelemetry/semconv/templating/markdown/__init__.py:42]: I'm not sure how I feel about this new class being "together" with the rest of the actual functionality of the generator. I wonder if it wouldn't be better if it was in a separated file for ex.

**Comment by Oberon00** [comment]
Please give the PR a more descriptive / concrete title, like "Add diff output on errors"

File: semantic-conventions/src/opentelemetry/semconv/templating/markdown/__init__.py (hunk 1 above)
Comment (joaopgrassi) [reply semantic-conventions/src/opentelemetry/semconv/templating/markdown/__init__.py:42]: I guess just a file on its own, TableDiffHelper or something :D

File: semantic-conventions/src/opentelemetry/semconv/templating/markdown/utils.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        Prints git-like colored diff using ANSI terminal coloring.
+        Diff is "from a to b", that is, red text is text deleted in `a`
+        while green text is new to `b`
+        """
+        if not os.environ.get("COLORED_DIFF"):
+            # return "\n".join(difflib.context_diff(a, b))
+            added = lambda text: f"added:'{text}'"
+            removed = lambda text: f"removed:'{text}'"
+        else:
+            added = lambda text: cls.colorize_text(0, 255, 0, text)
```
Comment (jsuereth) [inline semantic-conventions/src/opentelemetry/semconv/templating/markdown/utils.py:67]: RGB is 6-bit not 8-bit, so I think any value > 64 doesn't matter.

File: semantic-conventions/src/tests/data/markdown/table_generation_conflict/expected-no-colors.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+**Status**: [Experimental](../../../document-status.md)
+
+**type:** `foo.bar`
+
+**Description:** Test case
+
+<!-- semconv redis -->
+
+| Atadded:'t'ributeremoved:' ' | Type   | Description added:' '| Examples | Requirement Level |
```
Comment (jsuereth) [inline semantic-conventions/src/tests/data/markdown/table_generation_conflict/expected-no-colors.md:11]: Am I missing something or are these `added:'t'` unexpected.

**Review by jsuereth — COMMENTED** [review]
Overall love this nice addition, but not sure I understand how the tests are functioning given the examples.

Also, as a side note, when I last did ANSI color, I stuck with the 255 index table using something like this: https://github.com/jsuereth/streamerz/blob/master/ansi/src/main/scala/com/jsuereth/ansi/Ansi.scala#L64

**Review by jsuereth — APPROVED** [review]

File: semantic-conventions/src/tests/data/markdown/table_generation_conflict/expected-no-colors.md (hunk 3 above)
Comment (jsuereth) [reply semantic-conventions/src/tests/data/markdown/table_generation_conflict/expected-no-colors.md]: This looks a bit more readable to me.

**Comment by joaopgrassi** [comment]
@hughesjj this is been open for much long.. would you have the time to fix the conflicts so we can merge this? I think it's a great addition.

**Review by joaopgrassi — APPROVED** [review]

**Pull request merged**